jobteaser/terraform-wrapper
# Upgraded from Claranet Branch + some zsh hacks

# So basically it's a 140 commits behind update + some hacks (for zsh only) required when you manage python on your own and do not use the system defaults (nixpkgs,pyenv,brew...)

- changed some checks
- doing some adjustments for zsh shell
- Prepare `v7.9.0`
- TER-455: Changelog update
- TER-455: Test both with Tenant ID or Sub ID
- TER-455: Raise exception if no arg passed
- TER-455: Some debug logs
- TER-455: Handle Azure session check via Tenant ID
- TER-458: terraform 0.13 no longer supports -v, use version subcommand
- TER-456: fix docstring
- TER-456: s/cwd/dir/g
- TER-456: Apply suggestion to bin/tfwrapper
- TER-456: Changelog update
- TER-456: UTs for `tfwrapper.get_stack_envvars`
- TER-456: Update current UTs to match new `foreach_select_stacks` return signature
- TER-456: Export TFWRAPPER env variables when using `foreach`
- TER-456: Load stack configurations in `foreach` command
- TER-456: Load stack configurations in `foreach` command
- Prepare `v7.8.0`
- TER-439: Upgrade boto/AWS dependencies too  to match common libraries with Azure ones
- TER-439: `requirements-azure.txt` fully regenerated by `pip-compile`  with all `azure-cli` dependencies included
- TER-439: Bump `azure-cli` to latest `2.6.0`
- TER-439: Bump `azure-cli` to latest `2.5.1`
- TER-454: use testpath to mock terraform invocations
- TER-454: support missing or empty state.yml loading tests
- TER-454: update state.yml loading tests for missing or empty config (failing)
- Prepare `v7.7.0`
- TER-453: Some more unit tests
- TER-453: Add `switchver` in args unit tests
- TER-453: Don't load stack config for wrapper local commands  which don't need terraform
- TER-451: update testing dependencies (most notably pook to remove python 3.8 deprecation warnings)
- TER-451: remove pip-tools from runtime dependencies and update dependencies with pip-tools 4.5.1
- TER-450: use proper python image version for each tox env
- Add python 3.6 to tested versions
- Add python 3.8 to tested versions
- Prepare `v7.6.1`
- TER-331: Allow `<patch>[0-9]` suffixes to custom provider release version
- Prepare `v7.6.0`
- TER-439: Bump `azure-cli` to `2.1.0`
- TER-444: let black's line-length setting match flake8's max-line-length
- TER-444: use tox -e lint in CI
- TER-444: use flake8/flake8-docstrings
- TER-444: add pre-commit hooks
- TER-443: Pass TF_PLUGIN_CACHE_DIR environment variable to terraform if available
- TER-271: add --http-cache-dir CLI argument to configure HTTP(S) requests cache directory (defaults to ~/.terraform.d/http-cache)
- TER-271: enable urllib3's retry logic (cannot be tested with pook) and cleanup requests session initialization
- TER-271: update sample Github response headers
- TER-271: cache HTTPS request for 15 minutes
- TER-271: use requests.Session() (still fails as caching is not yet implemented)
- TER-271: add testcase for request caching (fails as caching is not yet implemented)
- TER-271: add unit tests for tfwrapper.search_on_github()
- TER-442: remove unused code and move GITHUB_RELEASES import where it is needed
- TER-442: improve code readability (wrap chained method invocations in parenthesis)
- TER-442: add link to black IDE integration doc
- TER-442: format code with black
- TER-347: Prepare `v7.5.1`
- TER-347: Update requirements
- Prepare `v7.5.0`
- TER-439 Change mgmt-storage version
- TER-439 Bump to latest version 77
- TER-439 Bump azure-cli to 2.0.75
- fix syntax check
- Add example for `kubeconfig_update` parameter
- Add table of contents to README.md
- Add tests for configuration parsking. Use caplog
- add doc
- improve log
- Better code for retrieving kubeconfig parameter
- Add new feature to refresh kubeconfig
- TER-423: Prepare `v7.4.1`
- TER-423: Fix directories path detection in Makefile
- TER-438 TER-343: Prepare `v7.4.0`
- TER-35: Prepare `v7.3.0`
- improve documentation for kubeconfig management
- Rename suffix to extension
- Add tests
- Add support for .tar.gz, .tar.bz2 (not tested) and version number starting with "v"
- TER-35: document foreach in README
- TER-35: fix foreach when -c is not specified
- TER-35: return code from invoked commands
- TER-35: use shell executable from SHELL environment variable if available
- TER-35: filter out stack configurations with no matching stack directory
- TER-35: pyflakes on tests
- TER-35: print help earlier and add tests
- TER-35: add lint env to tox (use with `tox -e lint`)
- TER-35: foreach shell execution
- TER-35: implement foreach
- TER-35: add foreach_select_stacks
- TER-35: add get_stack_from_config_path
- TER-35: extract get_stack_config_filename
- TER-35: extract get_stack_dir and get_stack_config_path
- TER-35: extract common pytest fixtures
- TER-35: detect_stack can now be invoked with `raise_on_missing=False` to not raise an error if any component required to identify a specific stack is missing
- TER-35: refactor detect_stack and load_wrapper_config
- TER-35: add tests for detect_stack
- TER-35: add tests for detect_config_dir
- TER-35: ignore command line arguments in tests (this allows to pass `--pdb` to pytest, e.g. `tox -- --pdb`
- TER-35: refactor config dir detection and wrapper config loading
- TER-35: add tests for load_wrapper_config()
- TER-35: add tox job as test stage
- TER-35: introduce unit tests
